pilot/pkg/model/validation.go

```diff
@@ -1547,27 +1547,41 @@ func checkServiceRoleBinding(in *rbac.ServiceRoleBinding) error {
 			errs = appendErrors(errs, fmt.Errorf("do not use * for names or not_names (in rule %d)", i))
 		}
 	}
-	if in.RoleRef != nil && len(in.Actions) > 0 {
-		errs = appendErrors(errs, fmt.Errorf("only one of `roleRef` or `actions` can be specified"))
+	moreThanOneRoleFieldError := "only one of `roleRef`, `role`, or `actions` can be specified"
+	if in.RoleRef != nil && in.Role != "" {
```

Comment:
Can we refactor to only check each field once?

```
cnt := 0
if in.RoleRef != nil {
  cnt += 1
}
if in.Role != "" {
 cnt += 1
}
if len(in.Actions) > 0 {
 cnt += 1
}
if cnt != 1 {
 // Error! one of roleRef, role and actions must be set
 return
}

Reply:
Very good trick. Changed it. Thanks!